src/software/thunderscope/robot_diagnostics/chicker.py

```diff
@@ -95,82 +90,32 @@ def change_button_state(self, button, enable):
-            print("Geneva:", self.geneva_value, "Power:", self.power_value)
-
-        # radio colors
-        if self.no_auto_button.isChecked():
-            self.radio_checkable["auto_kick"] = True
-            self.radio_checkable["auto_chip"] = True
-            if self.radio_checkable["no_auto"]:
-                print("No Auto clicked")
-                print("Geneva:", self.geneva_value, "Power:", self.power_value)
-            self.radio_checkable["no_auto"] = False
-            self.change_button_state(self.charge_button, True)
-            if self.charged:
-                self.change_button_state(self.chip_button, True)
-                self.change_button_state(self.kick_button, True)
-
-        elif self.auto_kick_button.isChecked():
-            self.radio_checkable["no_auto"] = True
-            self.radio_checkable["auto_chip"] = True
-            if self.radio_checkable["auto_kick"]:
-                print("Auto Kick clicked")
-                print("Geneva:", self.geneva_value, "Power:", self.power_value)
-            self.radio_checkable["auto_kick"] = False
-            self.change_button_state(self.chip_button, False)
-            self.change_button_state(self.kick_button, False)
-            self.change_button_state(self.charge_button, False)
+        geneva_value = self.geneva_slider.value()
+        self.geneva_label.setText(Slot.Name(geneva_value))
+
+        power_value = self.power_slider.value()
+        self.power_label.setText(str(power_value))
+
+        # if autokick is enabled, we don't want to allow kick/chip
+        auto_kick_enabled = (
+            self.auto_kick_button.isChecked() or self.auto_chip_button.isChecked()
+        )
+        self.change_button_state(self.kick_button, not auto_kick_enabled)
+        self.change_button_state(self.chip_button, not auto_kick_enabled)
+
+        power_control = PowerControl()
+        power_control.geneva_slot = geneva_value
 
+        # If auto is enabled, we want to populate the autochip or kick message
+        if self.auto_kick_button.isChecked():
+            power_control.chicker.auto_chip_or_kick.autokick_speed_m_per_s = power_value
         elif self.auto_chip_button.isChecked():
-            self.radio_checkable["no_auto"] = True
-            self.radio_checkable["auto_kick"] = True
-            if self.radio_checkable["auto_chip"]:
-                print("Auto Chip clicked")
-                print("Geneva:", self.geneva_value, "Power:", self.power_value)
-            self.radio_checkable["auto_chip"] = False
-            self.change_button_state(self.chip_button, False)
-            self.change_button_state(self.kick_button, False)
-            self.change_button_state(self.charge_button, False)
+            power_control.chicker.auto_chip_or_kick.autochip_distance_meters = (
+                power_value
+            )
+        elif self.no_auto_button.isChecked():
+            pass
+            # TODO ATTACH A CALLBACK AND SENDPROTO/KICK IMMEDIATELY
```

Comment:
nit: add issue number
Also, not sure about this todo. If we don't want auto (assuming auto kick), we want to kick immediately?